### PR TITLE
add FakeDLEq; add pure go + darwin build commands

### PR DESCRIPTION
**Pull request by noot**
this PR attempts to fix the build for non-linux platforms by removing the Rust dependency completely.

it adds make commands `make build-go` which builds the pure go project and `make build-go-darwin` which builds the pure go project for `GOOS=darwin` and `GOARCH=arm64`.

**WARNING**: the pure go build should not be used in reality, as the DLEq prover has been stubbed out and now proves nothing.

### CODE REVIEW

File: monero/wallet_client.go [hunk 1]
```diff
@@ -496,7 +496,7 @@ func launchMoneroWalletRPCChild(walletRPCBin string, walletRPCBinArgs ...string)
 	cmd.Env = append(os.Environ(), "LANG=C", "LC_ALL=C", "TERM=dumb")
 
 	cmd.SysProcAttr = &syscall.SysProcAttr{
-		Pdeathsig: syscall.SIGTERM,
+		//Pdeathsig: syscall.SIGTERM,
```
Comment (noot) [inline monero/wallet_client.go:499]: @dimalinux I had to comment this out since it doesn't build for `GOOS=darwin`

Comment (dimalinux) [reply monero/wallet_client.go:499]: The goal here is that if `swapd` (or a unit test for that matter) is killed in a way that doesn't shut down the wallet, the Linux OS will send SIGTERM to the `monero-wallet-rpc` process. You can read more about it under `PR_SET_PDEATHSIG` in the `prctl` man page. We could separate out the `SysProcAttr` handling into different files with different OS build tags.

**Comment by dimalinux** [comment]
> WARNING: the pure go build should not be used in reality, as the DLEq prover has been stubbed out and now proves nothing.

What is the goal if the build result is dangerous? How do we prevent people from using it once the build is fixed?

File: Makefile [hunk 2]
```diff
@@ -61,6 +61,15 @@ install: init
...
 	./scripts/build.sh
 
+.PHONY: build-go
+build-go:
+	go build ./cmd/swapd
+	go build ./cmd/swapcli
+
+.PHONY: build-go-darwin
+build-go-darwin:
+	GOOS=darwin GOARCH=arm64 $(MAKE) build-go
```
Comment (dimalinux) [inline Makefile:71]: I'm a little confused.  `build`, `build-all`, `build-go` and `build-go-darwin` are all go specific, but the `build-*` targets depend on `init` which builds the rust code. Which targets are supposed to be setting `cgodleq`?

Comment (noot) [reply Makefile:71]: yeah, I agree this is confusing, I updated the tags so `cgodleq` is default again, and `build-go` and `build-go-darwin` use `-tags=fakedleq`

Comment (dimalinux) [reply Makefile:71]: I don't see anyone setting `-tags=fakedleq`. Did you push that change? If you want me to set an environment variable here in the Makefile and update the script to use it, Iet me know.

Comment (noot) [reply Makefile:71]: it didn't push for some reason, updated

**Review by dimalinux — APPROVED** [review]
If these targets are needed longer term, we should rename them to something like `build-fakedleq` and `build-fakedleq-darwin` to decrease the chance of someone getting confused and thinking that the resulting executables are safe to perform swaps with.

**Pull request merged**